### PR TITLE
content/wip/links-to-EDU-form

### PR DESCRIPTION
**Pull request by peggygannon**

### CODE REVIEW

**Comment by peggygannon** [comment]
@J-Quag @iantopper EDU link PR ready for your review.

**Review by bethpotts — APPROVED** [review]
Looks great! I made a couple little edits. Thanks for getting this done so fast, Peggy.

**Comment by rroueche** [comment]
@J-Quag technically the changes look good to me, but I have a few comments for you -- if you like these, then we can update and re-review:

1) On the [Fry Scholarship page](https://vetsgov-pr-5608.herokuapp.com/education/gi-bill/survivors-dependent-assistance/fry-scholarship/), I think we can go directly to the form intro page itself (instead of the generic apply page) because the form intro page lists out the steps for how to prepare and check eligibility etc: https://www.vets.gov/education/apply-for-education-benefits/application/5490/introduction

2) On the [Transfer page](https://vetsgov-pr-5608.herokuapp.com/education/gi-bill/transfer/), I think we can also link directly to the form into page: https://www.vets.gov/education/apply-for-education-benefits/application/1990e/introduction

3) On the Montgomery Pages (https://vetsgov-pr-5608.herokuapp.com/education/gi-bill/montgomery-active-duty/) and (https://vetsgov-pr-5608.herokuapp.com/education/gi-bill/montgomery-selected-reserve/), can we also go directly to the form form intro? https://www.vets.gov/education/apply-for-education-benefits/application/1990e/introduction

4) On the [Dependents Page](https://vetsgov-pr-5608.herokuapp.com/education/gi-bill/survivors-dependent-assistance/dependents-education/) can we also go directly to the form intro: https://www.vets.gov/education/apply-for-education-benefits/application/5490/introduction

**Comment by bethpotts** [comment]
@J-Quag @iantopper I see a couple more tweaks from the editorial side. Letting you know that I'm going back in to make a few more commits.

**Comment by bethpotts** [comment]
@J-Quag If you agree with @rroueche that the above pages she lists should point directly to the form intro rather than the general application process page, the content team can implement. Let me know! I like the idea of fewer steps for the Veteran to take, so I'll throw in a vote for directing users to the form itself.

**Comment by J-Quag** [comment]
So part of what we're creating for Education is the wizard that will allow users to go through a flow that indicates what form they should fill out. The thought is that users can go through that if they are not definitely sure of the form they should fill out. 
Though if they are in the discovery pages, I would suppose that is being worked out. @goldenmeanie  do you have thoughts on this? 
If they are in pages for discovery and that is the form they want, we could just direct them to that form instead of the apply page w/wizard?

**Comment by rroueche** [comment]
@J-Quag i still think a forms wizard could be a valuable tool -- probably in addition to the eligibility/discovery static content. I think we can link directly to the forms from the discovery content, but that shouldn't detract from the wizard work. Maybe part of the wizard discovery could be testing out how people react to reading to find the right tool for them, vs using a wizard to be directed to the right one.

**Comment by goldenmeanie** [comment]
Some of those pages would need to link to multiple forms, right? For example, GI Bill is either 1990, 1990E, or 1995.

**Comment by rroueche** [comment]
@goldenmeanie @J-Quag there are only a few places where I recommend linking directly, see recs 1-4 in my comment above and let me know what ya think!

**Comment by rroueche** [comment]
Related to: https://github.com/department-of-veterans-affairs/vets.gov-team/issues/2934

**Comment by goldenmeanie** [comment]
I am totally on-board with the ability to apply from these pages! 

On the general apply page we have an "option button" that gives a little help in choosing the right form. We could embed that button on the benefit pages and I think it would cover the cases where there are multiple forms you might need. Does that make sense?

**Comment by rroueche** [comment]
@goldenmeanie i think that is a fabulous idea! @bethpotts or @amybehrens sound good to you? If so, then we can move fwd w/that plan -- @goldenmeanie may need your help in terms of the show/hide options thing you mentioned on the phone.

**Comment by bethpotts** [comment]
@goldenmeanie @rroueche @peggygannon Sounds like a great idea! Peggy has already updated the links on the pages you listed earlier in this string where there's no need for the option button. The links go directly to the appropriate intro page for the form itself.

**Comment by leannammiller** [comment]
@peggygannon it seems like this is complete? if so please close!

**Comment by peggygannon** [comment]
@leannammiller Closing this. Thanks!

**Comment by peggygannon** [comment]
@leannammiller

File: content/pages/education/other-educational-assistance-programs/call-to-service.md [hunk 1]
```diff
@@ -44,6 +44,8 @@ You may chose from the following:
 
 ### How do I get these benefits?
 
+Apply online with VA Form 22-1990N. [Find out how to apply](/education/apply/).
```
Comment (bethpotts) [inline content/pages/education/other-educational-assistance-programs/call-to-service.md]: @goldenmeanie You're right! You found another one. I see this wasn't in Rachael's original list. Yes, please go ahead and think this one directly too. Thanks!!

Comment (bethpotts) [inline content/pages/education/other-educational-assistance-programs/call-to-service.md]: @goldenmeanie Forgot to say, please link the whole line as Rachael did above in her last couple of commits: [Apply online with VA Form 22-1990N]. Thanks!

**Comment by bethpotts** [comment]
@goldenmeanie @J-Quag We're ready to get this merged and to production tomorrow or Wednesday if you want to take a last look. I'll ask Chris V. to review our markdown to make sure all's well if that sounds good to you. Then Sophia can merge for us. Thanks! It'll be great to wrap this one up.

**Comment by peggygannon** [comment]
@cvalarida This pull request resolves this issue: department-of-veterans-affairs/vets.gov-team/issues/4331.

It's a long PR (sorry! it's been around a while) Can you review the code in the most recent commits?

**Review by cvalarida — CHANGES_REQUESTED** [review]
Still need to hook up the JS so the wizard actually works, so I'm blocking this from getting merged until I (or somebody) can get that fix in.

File: src/js/no-react-entry.js [hunk 2]
```diff
@@ -1,6 +1,8 @@
 import createCommonStore from './common/store';
 import createLoginWidget from './login/login-entry';
 
+const wizardPages = new Set(['/education/apply/', '/education/eligibility/']);
```
Comment (cehsu) [inline src/js/no-react-entry.js]: The wizard is now required on the eligibility page.

Comment (peggygannon) [inline src/js/no-react-entry.js]: @cehsu @cvalarida Hello! Looks like the wizard is enabled. Is this PR ready to be merged?

File: src/js/no-react-entry.js [hunk 3]
```diff
@@ -16,7 +18,7 @@ require('./legacy/mega-menu.js');
 // New sidebar menu
 require('./legacy/sidebar-navigation.js');
 
-if (location.href.indexOf('education/apply') >= 0) {
+if (wizardPages.has(location.pathname)) {
```
Comment (cvalarida) [inline src/js/no-react-entry.js]: Ooh, this is much nicer!

**Review by cvalarida — APPROVED** [review]
Tested locally and it works. I think we're golden!

**Comment by cvalarida** [comment]
Ideally, we wouldn't duplicate all that html, but I this this is fine for now. We can circle back around to it and come up with a way to maybe provide a `div` with a specific ID that gets populated with the elements when the wizard javascript file is loaded. This would reduce duplication, though it's a little nasty. I'd rather use the templating engine, but... :man_shrugging:

**Comment by cehsu** [comment]
Great idea @cvalarida, @peggygannon is there any reason to keep this as HTML? Would content/design ever want to be able to edit this directly or now that it's fairly done, is it ok to simply handle this using JavaScript?

**Comment by cvalarida** [comment]
More specifically, is there a reason we'd need to have different content for this in the eligibility vs application process page?

**Comment by peggygannon** [comment]
@cehsu @cvalarida Good question! The only reason I would encourage to keep this wizard in HTML is so content and design would still have the flexibility to edit the wizard questions if needed in the future. 
The content on the eligibility page should be the same as the apply page. 

 ccing @bethpotts  for her input as well. Beth, see Claire's question ^^ about Javascript.

**Comment by bethpotts** [comment]
@cehsu @cvalarida @peggygannon  What's the benefit to switching to javascript? I guess I hesitate to do that if we wouldn't be able to edit the words further--if we did testing, for instance, and found that we wanted to reword some questions. But would love to understand why you'd like to move from html to javascript. I don't have any insight there... thanks!
Tagging @goldenmeanie in case he has any thoughts on this. I think this option button for the application process page started with you, Alex, right?

**Comment by goldenmeanie** [comment]
My read on this is that the plan is to use Javascript to dynamically insert a snippet of HTML into all the places on vets.gov that it shows up. The idea being that if there are any edits you only make them in the one place and it ripples out from there. But maybe I am misinterpreting the intent?

I would generally prefer for stuff like this to remain editable by mere mortals. But I would love to hear more from the devs.

**Comment by cvalarida** [comment]
@goldenmeanie @peggygannon @bethpotts

I'm not sure if the content team has any access restrictions or anything like that, but if we move it to javascript, you guys should still be able to edit it just fine. The main benefit for inserting the html via javascript is, like @goldenmeanie said, so we can edit the content in one file and have the changes cascade to everywhere it's used.

Also, if we found we wanted to add it somewhere else, we can just "plug it in" and have it just work.

I just talked to @cehsu and it sounds like she'll get started on that conversion. It shouldn't take long, and there _shouldn't_ (:crossed_fingers:) be a problem with editing the content the way we're looking at wiring this up.

More information to come!

**Comment by cvalarida** [comment]
**Update:** We'll go ahead and merge this now, while @cehsu works on the unified, plug-n-play version.

**Pull request merged**